### PR TITLE
Allow `RandomCutMix` to use conditional pair sampling like `RandomMixUp`.

### PR DESCRIPTION
**Pull request by MylesBartlett**
- Allow  to use conditional pair sampling like `RandomMixUp`.
- Add `deprecated` decorator.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,35 @@
...
+    | _build
+    | buck-out
+    | build
+    | dist
+    | mypy.ini
+    | results
+  )/
+)
+'''
+extend_exclude = ".undodir/*"
```
Comment (tmke8) [inline pyproject.toml:32]: do you need `exclude` if you have `extend_exclude`?

File: ranzen/torch/transforms/cutmix.py [hunk 2]
```diff
@@ -31,10 +32,10 @@ def __init__(
         self,
         alpha: float = 1.0,
         *,
-        p: float = 0.5,
+        p: float = 1.0,
         num_classes: int | None = None,
         inplace: bool = False,
-        seed: Optional[int] = None,
+        generator: torch.Generator | None = None,
```
Comment (tmke8) [inline ranzen/torch/transforms/cutmix.py:38]: This union operator will make it not usable with hydra, right?

File: tests/decorator_test.py [hunk 3]
```diff
@@ -18,46 +18,44 @@ def func(self) -> None:
         def no_docstring(self) -> None:
             pass
 
-    class CorrectImplementation(BaseClass):
+    class CorrectImplementation(BaseClass):  # pyright: ignore
```
Comment (tmke8) [inline tests/decorator_test.py:21]: what's pyright's complaint here? we could maybe delete this test alltogether, because we deprecated `@implements`.

**Review by tmke8 — APPROVED** [review]

File: ranzen/torch/transforms/cutmix.py (hunk 2 above)
Comment (tmke8) [reply ranzen/torch/transforms/cutmix.py:38]: (on python 3.8)

Comment (MylesBartlett) [reply ranzen/torch/transforms/cutmix.py:38]: hmm this is true

File: tests/decorator_test.py (hunk 3 above)
Comment (MylesBartlett) [reply tests/decorator_test.py:21]: that the class is unused (I tried to specify the error to be ignore but had no luck getting it to work)

Comment (MylesBartlett) [reply tests/decorator_test.py:21]: Let's just leave the test there for now; it's not doing any harm

File: pyproject.toml (hunk 1 above)
Comment (MylesBartlett) [reply pyproject.toml:32]: 🤷

Comment (MylesBartlett) [reply pyproject.toml:32]: Seems not so I'll just remove it

**Pull request merged**